nixos/modules/services/hardware/spacenavd.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.hardware.spacenavd;
+
+in {
+
+  options = {
+    hardware.spacenavd = {
+      enable = mkEnableOption "spacenavd to support 3DConnexion devices";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.spacenavd = {
```

Comment:
Could we use the upstream file and only patch the things that need changing? https://github.com/FreeSpacenav/spacenavd/blob/master/contrib/systemd/spacenavd.service

Reply:
The file is not maintained by upstream though (see https://github.com/FreeSpacenav/spacenavd/blob/master/contrib/README). Also it is so small, that I don't see the benefit of patching it. Maybe if there are some relevant developments in the future that could be considered.